src/components/GenericDropdown.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useRef } from 'react';
+import DropdownList, { DropdownItem } from '../components/DropdownList';
+import styled from 'styled-components';
+import useComponentVisible from '../hooks/useComponentVisible';
+
+interface Props {
+	buttonDisplay: React.ReactNode;
+	dropdownItems: DropdownItem[];
+}
+
+const DropdownButton = styled.button`
+	display: flex;
+	justify-content: space-between;
+	align-items: center;
+	border: 1px, solid, black;
+`;
+
+const Container = styled.div`
+	display: flex;
+	flex-direction: column;
+`;
+
+function GenericDropdown({ buttonDisplay, dropdownItems }: Props) {
+	const ref = useRef<HTMLDivElement>(null);
+	const { isComponentVisible, setIsComponentVisible } = useComponentVisible(
+		ref,
+		true
+	);
+	const handleClick = () => {
+		setIsComponentVisible(!isComponentVisible);
+	};
+
+	return (
+		<Container ref={ref}>
+			{
+				<DropdownButton onClick={handleClick}>
+					{buttonDisplay}
+					<i
+						className={
+							isComponentVisible
+								? 'fas fa-angle-up'
+								: 'fas fa-angle-down'
+						}
+					/>
```

Comment:
Cool use of icons! :)

Reply:
This is super small and might not be worth the effort - but is it possible to make an animation to switch between both icons?

Reply:
That was a feature I was thinking about adding but totally forgot about it haha, I'll try looking into that again!

Reply:
I would say let's not do it as part of this ticket. Transitions are a bit of work in React, and I think it will be something we add later. Nice catch though Isaiah!